tests/csm/rest/test_iam_users.py

```diff
@@ -1430,3 +1430,469 @@ def test_37783(self):
+        assert_utils.assert_true(resp.status_code == HTTPStatus.CREATED, "IAM user creation failed")
+        uid = resp.json()["tenant"] + "$" + payload['uid']
+        self.log.info("STEP 2: Perform get iam users")
+        get_resp = self.csm_obj.get_iam_user(uid)
+        assert_utils.assert_true(get_resp.status_code == HTTPStatus.OK, "Get IAM user failed")
+        self.log.info("STEP 3: Update any random parameters for created user")
+        payload = self.csm_obj.iam_user_patch_random_payload()
+        self.log.info("Random payload is %s:", payload)
+        resp1 = self.csm_obj.modify_iam_user_rgw(uid, payload)
+        assert_utils.assert_true(resp1.status_code == HTTPStatus.OK, "IAM user modify failed")
+        self.log.info("STEP 4: Perform get iam users to verify updated random parameters")
+        get_resp = self.csm_obj.get_iam_user(uid)
+        assert_utils.assert_true(get_resp.status_code == HTTPStatus.OK, "Get IAM user failed")
+        self.log.info("[END]Update request with uid and other parameters randomly. ")
+        self.log.info("##### Test completed -  %s #####", test_case_name)
+
+    @pytest.mark.csmrest
+    @pytest.mark.lc
+    @pytest.mark.cluster_user_ops
+    @pytest.mark.parallel
+    @pytest.mark.tags('TEST-38093')
+    def test_38093(self):
+        """
+        Verify PATCH iam user request for Invalid key type
+        """
+        test_case_name = cortxlogging.get_frame()
+        self.log.info("##### Test started -  %s #####", test_case_name)
+        self.log.info("[START] Creating IAM user")
+        uid = "iam_user_1_" + str(int(time.time_ns()))
+        self.log.info("Creating new iam user %s", uid)
+        payload = self.csm_obj.iam_user_payload_rgw("loaded")
+        payload.update({"uid": uid})
+        payload.update({"display_name": uid})
+        resp = self.csm_obj.create_iam_user_rgw(payload)
+        self.log.info("Verify Response : %s", resp)
+        assert_utils.assert_true(resp.status_code == HTTPStatus.CREATED, "IAM user creation failed")
+        uid = payload["tenant"] + "$" + uid
+        self.created_iam_users.add(uid)
+        invalid_keys = ["s3swift", "123", None, "", "@#$", "null"]
+        for key_value in invalid_keys:
+            self.log.info("Testing for key value %s", key_value)
+            payload = {}
+            payload.update({"key_type": key_value})
+            resp = self.csm_obj.modify_iam_user_rgw(uid, payload)
+            self.log.info("Verify Response : %s", resp)
+            assert_utils.assert_true(resp.status_code == HTTPStatus.BAD_REQUEST,
+                                     "Patch request status code failed")
+            if CSM_REST_CFG["msg_check"] == "enable":
+                assert_utils.assert_true(resp.json()["message"] ==
+                                         self.rest_resp_conf[4099]['InvalidKeyType'][0]
+                                         , "Response message check failed")
+
```

Comment:
add  self.log.info("##### Test completed -  %s #####", test_case_name) for all newly added tests

Reply:
Adding it.

Reply:
done